cpp/src/arrow/pretty_print.cc

```diff
@@ -82,7 +82,9 @@ void PrettyPrinter::OpenArray(const Array& array) {
 void PrettyPrinter::CloseArray(const Array& array) {
   if (array.length() > 0) {
     indent_ -= options_.indent_size;
-    Indent();
+    if (!options_.skip_new_lines) {
+      Indent();
+    }
```

Comment:
This is to fix a bug in printing arrays that lead to
```
[10,20,30,40   ]
```

instead of 

```
[10,20,30,40]
```

when `skip_new_lines=True`